### PR TITLE
Add GA events to KVM details + store NUMA view toggle state in local storage

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added GA events to KVM details (NUMA view toggle, "Show more/less storage pools", "Show more/less NUMA nodes")
- Stored NUMA view toggle state in local storage, on a per-KVM basis

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Load a KVM details page and check that the aggregate view shows by default.
- Toggle to NUMA node view and refresh that page.
- Check that the page is still set to NUMA node view.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2080
Fixes canonical-web-and-design/MAAS-squad#2086

### CODE REVIEW

**Review by anthonydillon — APPROVED** [review]
Thanks @Caleb-Ellis LGTM :+1:

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMNumaResources/KVMNumaResources.tsx [hunk 1]
```diff
@@ -45,7 +46,14 @@ const KVMNumaResources = ({ numaNodes }: Props): JSX.Element => {
             appearance="base"
             data-test="show-more-numas"
             hasIcon
-            onClick={() => setExpanded(!expanded)}
+            onClick={() => {
+              setExpanded(!expanded);
+              sendAnalyticsEvent(
```
Comment (anthonydillon) [inline ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMNumaResources/KVMNumaResources.tsx:51]: We should check analytics is enabled?

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMStorage/KVMStorage.tsx [hunk 2]
```diff
@@ -85,7 +86,16 @@ const KVMStorage = ({ id }: Props): JSX.Element | null => {
               appearance="base"
               data-test="show-more-pools"
               hasIcon
-              onClick={() => setExpanded(!expanded)}
+              onClick={() => {
+                setExpanded(!expanded);
+                sendAnalyticsEvent(
```
Comment (anthonydillon) [inline ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMStorage/KVMStorage.tsx:91]: Check analytics is enabled first.

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx [hunk 3]
```diff
@@ -82,7 +84,11 @@ const KVMSummary = (): JSX.Element => {
   const pod = useSelector((state: RootState) =>
     podSelectors.getById(state, Number(id))
   );
-  const [viewByNuma, setViewByNuma] = useState(false);
+  const [viewByNuma, setViewByNuma] = useStorageState(
```
Comment (anthonydillon) [inline ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx:87]: Is `viewByNuma` required?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Sorry found some issues in the code.

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMNumaResources/KVMNumaResources.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMNumaResources/KVMNumaResources.tsx:51]: Yes, we should!

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx (hunk 3 above)
Comment (Caleb-Ellis) [reply ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx:87]: Yeah because it determines whether the aggregate or NUMA view is shown.

**Comment by Caleb-Ellis** [comment]
> Sorry found some issues in the code.

Thanks @anthonydillon updated the PR to check if analytics are enabled first :+1:

**Review by anthonydillon — APPROVED** [review]
Thanks @Caleb-Ellis

**Pull request merged**